src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -477,6 +479,12 @@ public Authentication authenticate(Authentication authentication)
                         UsernamePasswordAuthenticationToken token = (UsernamePasswordAuthenticationToken) authentication;
                         GithubAuthenticationToken github = new GithubAuthenticationToken(token.getCredentials().toString(), getGithubApiUri());
                         SecurityContextHolder.getContext().setAuthentication(github);
+
+                        SecurityListener.fireAuthenticated(new GithubOAuthUserDetails(token.getName(), github.getAuthorities()));
+
+                        //TODO do we want to trigger a loggedIn event when using GitHub token ?
+                        // SecurityListener.fireLoggedIn(token.getName());
```

Comment:
Can the token be used to log in from the UI? I'll do some testing to figure out when this code is called. `SecurityListener#loggedIn` JavaDoc says:
> Fired when a user has logged in via the web UI.

Reply:
In the use of GitHub login, it's when we use the process showing you the GitHub login page etc. It's the case of "Web UI" authentication.
But in the case of GitHub token, we are in the same case as for the API Token in core, it's not trough Web UI but by REST calls, hence my TODO.

FTR I proposed changes in the JavaDoc of `loggedIn` in [#3074](https://github.com/jenkinsci/jenkins/pull/3074/files#diff-2a0e1f5e44d0d8701c5b9c0e0ab44ad9R65) from "via the web UI" to "notion of cache / storage of authentication"

Reply:
Ok, thanks for the link to the proposed JavaDoc changes. I looked through other `SecurityRealm` implementations (listed [here](https://jenkins.io/doc/developer/extensions/jenkins-core/#securityrealm)), but didn't find any that called `loggedIn` for a token-based authentication. The main reason to call fireLoggedIn would be to populate the last granted authorities, right?

Reply:
Essentially yes, but as you mentionned it's not a behavior that is really desired. I will remove the comments 👍